### DISCUSSION TITLE
Render new rows on scroll

### DISCUSSION THREAD

**Discussion by igorpost92** [petyosi/react-virtuoso#389] [Q&A]
Hi.
Is it possible to make virtuoso render new rows during the scroll?
Right now it renders new rows only when scroll gets close to the latest. I would like to render a few new rows before and after current rows during the scroll (like it works in react-vitrualized for example). Because i have complex rows and when user scrolls very fast there are blank spaces.

**Comment by petyosi** [comment]
No, this is not supported currently. Quite a few people have asked about it, I am open to PR on it, although I would personally not recommend using that approach.

**Reply by igorpost92** [reply]
Is there any way to avoid these blanks when user scrolls very fast?

**Reply by petyosi** [reply]
You can try using the scroll seek placeholders which allow rendering of lighter components when scrolling fast.

**Reply by igorpost92** [reply]
But i have cards of various heights and these cards will be jumping when user stops the scroll

**Reply by petyosi** [reply]
In that case, you can try implementing the viewport boundaries increase and see if it works for you. If it does, I am open to a PR.

**Comment by petyosi** [comment]
I've recently introduced a property that can address that - `increaseViewportBy` (working like overscan in react-virtualized).

**Reply by lexahall** [reply]
Does this property not apply for VirtuosoGrid? Is there any equivalent?

**Reply by lexahall** [reply]
@petyosi is there any support for this with Grids? I'm planning to try just implementing my grid as a list of rows instead, so that I can use the `increaseViewportBy`, but just want to make sure I didn't overlook something in the docs.

**Reply by petyosi** [reply]
@lexahall I don't provide free support over GH issues and discussions. Everything which the library supports is documented. If something is not there, you can read the source code. If something does not work as you expect, you can submit a PR or sponsor the development of the feature.

**Reply by lexahall** [reply]
Sure sure, I didn't see anything in the docs or source code for this. So I can definitely look into adding support for `increaseViewportBy` for Grids. I just wanted to make sure I wasn't misunderstanding anything before taking that next step. Thanks for the quick response - I really appreciate all the work you've done here.

**Reply by igorpost92** [reply]
Hi. I noticed, that increaseViewportBy takes size in pixels, while overscan in both react-virtualized and react-window takes items amount.
What the difference between increaseViewportBy and overscan in react-virtuoso?
